### ISSUE TITLE
Cannot read property 'getRules' of undefined after update to 8.3.0

### ISSUE THREAD

**Issue by creativeux** [AtomLinter/linter-eslint#1021]
## Issue Type
Bug

## Issue Description
Updated to 8.3.0 and Atom 1.20.0, and I am now getting ESLint errors in the console showing...

```
Cannot read property 'getRules' of undefined TypeError: Cannot read property 'getRules' of undefined
    at updateFixableRules (/Users/aastone/.atom/packages/linter-eslint/src/worker.js:26:9)
    at lintJob (/Users/aastone/.atom/packages/linter-eslint/src/worker.js:61:3)
    at process.<anonymous> (/Users/aastone/.atom/packages/linter-eslint/src/worker.js:101:22)
    at emitTwo (events.js:111:20)
    at process.emit (events.js:191:7)
    at process.nextTick (internal/child_process.js:752:12)
    at _combinedTickCallback (internal/process/next_tick.js:67:7)
    at process._tickCallback (internal/process/next_tick.js:98:9)
```
I've updated on multiple machines with the same errors. Did not fail on previous version. My project has `.eslintrc` and `.eslintignore` files, but removing them has no effect.

## Bug Checklist
- Restart Atom
- Verify the `eslint` CLI gives the proper result, while `linter-eslint` does not
- Paste the output of the `Linter Eslint: Debug` command from the Command Palette below

NOTE: Uninstalled and reinstalled `linter-eslint` a couple of times debugging.

```
Atom version: 1.20.0
linter-eslint version: 8.3.0
ESLint version: 3.19.0
Hours since last Atom restart: 0
Platform: darwin
Using local project ESLint from: /Users/aastone/Development/design-system/atom-button/node_modules/eslint
Current file's scopes: [
  "source.js"
]
linter-eslint configuration: {
  "lintHtmlFiles": false,
  "useGlobalEslint": false,
  "showRuleIdInMessage": true,
  "disableWhenNoEslintConfig": true,
  "eslintrcPath": "",
  "globalNodePath": "",
  "advancedLocalNodeModules": "",
  "eslintRulesDir": "",
  "disableEslintIgnore": false,
  "disableFSCache": false,
  "fixOnSave": false,
  "scopes": [
    "source.js",
    "source.jsx",
    "source.js.jsx",
    "source.babel",
    "source.js-semantic"
  ],
  "rulesToSilenceWhileTyping": [],
  "rulesToDisableWhileFixing": [],
  "ignoreFixableRulesWhileTyping": false
}
```

**Comment by creativeux** [comment]
Did some more digging, and it looks like updating my installed ESLint (3.19.0 -> 4.7.0) fixed the problem. Hopefully that helps in case it is actually a 3.x issue.

**Comment by Arcanemagus** [comment]
I'll get a fix out for this shortly, thanks for reporting this!

**Comment by MegrezZhu** [comment]
same problem here after updating to 8.3.0 but my ESLint is 4.7.0

**Comment by Arcanemagus** [comment]
@MegrezZhu Can you run the debug command to check which ESLint version is actually being used? This is supported from the very first version of v4.

**Comment by MegrezZhu** [comment]
@Arcanemagus 
```
Atom version: 1.20.0
linter-eslint version: 8.3.0
ESLint version: 3.19.0
Hours since last Atom restart: 0
Platform: win32
Using local project ESLint from: E:\Development\HBsurprise\client\node_modules\eslint
Current file's scopes: [
  "source.js"
]
linter-eslint configuration: {
  "lintHtmlFiles": false,
  "useGlobalEslint": false,
  "showRuleIdInMessage": true,
  "disableWhenNoEslintConfig": true,
  "eslintrcPath": "",
  "globalNodePath": "",
  "advancedLocalNodeModules": "",
  "eslintRulesDir": "",
  "disableEslintIgnore": false,
  "disableFSCache": false,
  "fixOnSave": false,
  "scopes": [
    "source.js",
    "source.jsx",
    "source.js.jsx",
    "source.babel",
    "source.js-semantic"
  ],
  "rulesToSilenceWhileTyping": [],
  "rulesToDisableWhileFixing": [],
  "ignoreFixableRulesWhileTyping": false
}
```

**Comment by Arcanemagus** [comment]
> ESLint version: 3.19.0
> Using local project ESLint from: E:\Development\HBsurprise\client\node_modules\eslint

@MegrezZhu Your project is still using v3.19.0 😉.

**Comment by MegrezZhu** [comment]
If I set "Use global ESLint installation" to yes, it's 4.7.0 (my globally installed version) but the bug still exists. And I check another project with locally installed ESLint of 4.7.0, the error is the same
```
Atom version: 1.20.0
linter-eslint version: 8.3.0
ESLint version: 3.19.0
Hours since last Atom restart: 0
Platform: win32
Using local project ESLint from: E:\Development\HBsurprise\client\node_modules\eslint
Current file's scopes: [
  "source.js"
]
linter-eslint configuration: {
  "lintHtmlFiles": false,
  "useGlobalEslint": false,
  "showRuleIdInMessage": true,
  "disableWhenNoEslintConfig": true,
  "eslintrcPath": "",
  "globalNodePath": "",
  "advancedLocalNodeModules": "",
  "eslintRulesDir": "",
  "disableEslintIgnore": false,
  "disableFSCache": false,
  "fixOnSave": false,
  "scopes": [
    "source.js",
    "source.jsx",
    "source.js.jsx",
    "source.babel",
    "source.js-semantic"
  ],
  "rulesToSilenceWhileTyping": [],
  "rulesToDisableWhileFixing": [],
  "ignoreFixableRulesWhileTyping": false
}
```
```
Atom version: 1.20.0
linter-eslint version: 8.3.0
ESLint version: 4.7.0
Hours since last Atom restart: 0.1
Platform: win32
Using global ESLint from: C:\Users\i4908\AppData\Roaming\npm\node_modules\eslint
Current file's scopes: [
  "source.js",
  "punctuation.terminator.statement.js"
]
linter-eslint configuration: {
  "useGlobalEslint": true,
  "lintHtmlFiles": false,
  "showRuleIdInMessage": true,
  "disableWhenNoEslintConfig": true,
  "eslintrcPath": "",
  "globalNodePath": "",
  "advancedLocalNodeModules": "",
  "eslintRulesDir": "",
  "disableEslintIgnore": false,
  "disableFSCache": false,
  "fixOnSave": false,
  "scopes": [
    "source.js",
    "source.jsx",
    "source.js.jsx",
    "source.babel",
    "source.js-semantic"
  ],
  "rulesToSilenceWhileTyping": [],
  "rulesToDisableWhileFixing": [],
  "ignoreFixableRulesWhileTyping": false
}
```

**Comment by Arcanemagus** [comment]
@MegrezZhu If you didn't restart Atom after changing that option and had it disabled for the first lint on a file it would still be using the original version, at least I'm guessing that is why it was still showing.

I manually verified the fix in #1024 works for this, getting #1019 merged and I'll push out a bugfix release.

**Comment by Arcanemagus** [comment]
v8.3.1 should fix this issue, thanks again for the report! (and get your ESLint versions updated 😛)
